src/app/page.tsx

```diff
@@ -12,18 +12,25 @@ export default async function Home() {
   const countryMapDataPromise = globalRepo.getMapDataForCountries();
   const disputedAreasPromise = globalRepo.getDisputedAreas();
   const ipcDataPromise = globalRepo.getIpcData();
-  const [countryMapData, disputedAreas, ipcData] = await Promise.all([
+  const nutritionDataPromise = globalRepo.getNutritionData();
+  const [countryMapData, disputedAreas, ipcData, nutritionData] = await Promise.all([
```

Comment:
@marinovl7 or @bohdangarchu could you take a look and confirm if the `app/page` is the right place for the `CountryNutrion` data fetching ? -> for me it made sense to implement it there cause all other map related basic data is loaded there as well

Reply:
looks good to me

Reply:
Actually, it seems to me that we can delay fetching the nutrition data until the user selects the nutrition map. It's my first time looking at this PR, so feel free to challenge me if you think I'm wrong :D

in my understanding, the nutrition data fetched here is passed to the Map component, which passes it to `VectorTileLayer` and `NutritionChoropleth`. `VectorTileLayer` just passes it to the methods of `MapOperations`, but those methods don't use it at all. So we don't need to pass it to this component at all.

`NutritionChoropleth` is only rendered when the user selects the nutrition map, so this component can fetch the data. If you use the `useNutritionQuery` hook in the component's body, it'll automatically be fetched the first time this component loads. But in this case, we have to do something with the loading state (tho this request should be really fast). But generally, we are trying to avoid loading data before it's absolutely needed, so I recommend using this approach.

The current implementation of the IPC data required the data to be there from the beginning, that's why it was fetched on the server. But this is also being reworked.